[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9724 / windows-2022_checkpart3_platform-support-windows](https://gradle-enterprise.elastic.co/s/3qyltggurs3fk)
- [elasticsearch-pull-request #83808 / part-3](https://gradle-enterprise.elastic.co/s/ks6jhkdemdqdm)
- [elasticsearch-pull-request #83416 / part-3](https://gradle-enterprise.elastic.co/s/uebeveapkr3dc)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT.test" -Dtests.seed=26C93011D1BE8C25 -Dtests.locale=om-KE -Dtests.timezone=Asia/Kuching -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from languages_mixed_numerics,colors,age* | mv_expand `language_code_short` | enrich languages_policy on language_name | rename hex_code AS `YXVtZpkG`, `language_code_short` AS `language_code_float` | sort language_code_integer DESC NULLS FIRST, YXVtZpkG ASC, language_name DESC NULLS FIRST, language_code_float NULLS LAST | rename `language_code_float` AS `language_code_double`| rename description as message | lookup join message_types_lookup on message | FORK  (where  NOT language_code_half_float == 50) (where true) (where false) | WHERE _fork == "fork2" | DROP _fork | mv_expand `language_code_integer` | where true OR true AND  NOT false OR  NOT false| rename language_code_byte as language_code | lookup join languages_lookup on language_code | sort language_code DESC, type ASC NULLS LAST | rename message AS age_range, language_code_scaled_float AS YXVtZpkG | limit 8555| rename age_range as message | lookup join message_types_lookup on message | grok mes
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test
- [main] 3 failures in test test (0.9% fail rate in 320 executions)
- [main] 2 failures in step part-3 (1.3% fail rate in 157 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 159 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -509,9 +509,6 @@ tests:
 - class: org.elasticsearch.xpack.sql.qa.mixed_node.SqlCompatIT
   method: testNullsOrderWithMissingOrderSupportQueryingNewNode
   issue: https://github.com/elastic/elasticsearch/issues/132249
-- class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/132273
 - class: org.elasticsearch.common.logging.JULBridgeTests
   method: testThrowable
   issue: https://github.com/elastic/elasticsearch/issues/132280

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -160,7 +160,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
         );
         if (outputValidation.success() == false) {
             for (Pattern allowedError : ALLOWED_ERROR_PATTERNS) {
-                if (allowedError.matcher(outputValidation.errorMessage()).matches()) {
+                if (isAllowedError(outputValidation.errorMessage(), allowedError)) {
                     return outputValidation;
                 }
             }
@@ -171,11 +171,22 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
 
     private void checkException(EsqlQueryGenerator.QueryExecuted query) {
         for (Pattern allowedError : ALLOWED_ERROR_PATTERNS) {
-            if (allowedError.matcher(query.exception().getMessage()).matches()) {
+            if (isAllowedError(query.exception().getMessage(), allowedError)) {
                 return;
             }
         }
         fail("query: " + query.query() + "\nexception: " + query.exception().getMessage());
+    }
+
+    /**
+     * Long lines in exceptions can be split across several lines. When a newline is inserted, the end of the current line and the beginning
+     * of the new line are marked with a backslash {@code \}; the new line will also have whitespace before the backslash for aligning.
+     */
+    private static final Pattern ERROR_MESSAGE_LINE_BREAK = Pattern.compile("\\\\\n\\s*\\\\");
+
+    private static boolean isAllowedError(String errorMessage, Pattern allowedPattern) {
+        String errorWithoutLineBreaks = ERROR_MESSAGE_LINE_BREAK.matcher(errorMessage).replaceAll("");
+        return allowedPattern.matcher(errorWithoutLineBreaks).matches();
     }
 
     @SuppressWarnings("unchecked")

